etna/lib/etna/clients/magma/formatting/models_csv.rb

```diff
@@ -0,0 +1,250 @@
+              yield "Model #{template.parent} is linked as a parent to #{template.name}, but it already has an attribute named #{template.name} #{parent_model_attribute_by_model_name.raw}." if block_given?
+            end
+
+            parent_model.template.build_attributes.build_attribute(template.name).tap do |attr|
+              attr.attribute_name = attr.name = template.name
+              attr.attribute_type = parent_link_type
+              attr.link_model_name = template.name
+              attr.desc = self.prettify(template.name)
+              attr.display_name = self.prettify(template.name)
+            end
+          end
+        end
+
+        def self.process_attribute(template:, attribute_name:, models:, row:, &err_block)
+          if template.nil?
+            yield "Found attribute #{attribute_name} but no model_name had been given!" if block_given?
+            return
+          end
+
+          attributes = template.build_attributes
+          existing_attribute = attributes.attribute(attribute_name)
+          if existing_attribute
+            if existing_attribute.attribute_type != AttributeType::COLLECTION
+              yield "Attribute #{attribute_name} of model #{template.name} has duplicate definitions!" if block_given?
+            end
+
+            attributes.raw.delete(attribute_name)
+          end
+
+          attributes.build_attribute(attribute_name).tap do |att|
+            COLUMNS_TO_ATTRIBUTES.each do |column, processor|
+              next unless (value = self.get_col_or_nil(row, column))
+              if processor.is_a?(Array)
+                processor, f = processor
+                value = f.call(value)
+              end
+
+              if !att.send(processor).nil? && !att.send(processor).empty?
+                yield "Value for #{processor} on attribute #{attribute_name} has duplicate definitions!" if block_given?
+              end
+
+              att.send(:"#{processor}=", value)
+            end
+
+            if att.attribute_type == AttributeType::LINK && !models.find_reciprocal(model_name: template.name, attribute: att)
+              models.build_model(att.link_model_name).build_template.build_attributes.build_attribute(template.name).tap do |rec_att|
+                rec_att.attribute_name = template.name
+                rec_att.display_name = self.prettify(template.name)
+                rec_att.desc = self.prettify(template.name)
+                rec_att.attribute_type = AttributeType::COLLECTION
+                rec_att.link_model_name = template.name
+              end
+            end
+          end
+        end
+
+        def self.prettify(name)
+          name.split('_').map(&:capitalize).join(' ')
+        end
+
+        def self.get_col_or_nil(row, col)
+          c = row[col]&.chomp
+          return nil if c&.empty?
+          c
+        end
+
+        def self.each_tsv_row(models = Models.new, model_keys = models.model_keys.sort, &block)
```

Comment:
Output function, yields rows of column values for each model, given by the ordering of model_keys.